lib/appmap/service/config_analyzer.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module AppMap
+  module Service
+    class ConfigAnalyzer
+      attr_reader :config_error
+
+      def initialize(config_file)
+        @config_file = config_file
+        @config = load_config
+      end
+
+      def app_name
+        @config.to_h[:name] if is_present?
+      end
+
+      def is_present?
+        File.exist?(@config_file)
+      end
+
+      def is_valid?
+        is_present? && @config.to_h.key?(:name) && @config.to_h.key?(:packages)
```

Comment:
Looking at this, I think that if we are going to indicate that the config file is not valid, we need to provide a reason. So, this should be added to the status spec.

The reasons I can think of:
```
1) File is not valid YAML
2) Name is missing
3) No packages are listed (but this is not really an error, the user may want to do that on purpose?)
4) The YAML is valid, but an error occurred loading the config file: {message}
```

Reply:
Will be implemented in `appmap-agent-validate`.